docs/source/format/Columnar.rst

```diff
@@ -765,6 +765,85 @@ application.
 We discuss dictionary encoding as it relates to serialization further
 below.
 
+.. _run-length-encoded-layout:
+
+Run-Length-encoded Layout
+-------------------------
+
+Run-Length is a data representation that represents data as sequences of the
+same a, called runs. Each run is represented as a value, and an integer
+describing how often this value is repeated.
+
+Any array can be run-length-encoded. A run-length encoded array has a single
+buffer holding as many 32-bit integers, as there are runs. The actual values are
+hold in a child array, which is just a regular array
+
+The dictionary is stored as an optional
+property of an array. When a field is dictionary encoded, the values are
+represented by an array of non-negative integers representing the index of the
+value in the dictionary. The memory layout for a dictionary-encoded array is
+the same as that of a primitive integer layout. The dictionary is handled as a
+separate columnar array with its own respective layout.
```

Comment:
Is this related? Or was this a copy-paste thing

Reply:
copy-paste mistake indeed